web/src/components/repo/pipeline/PipelineLog.vue

```diff
@@ -20,6 +20,13 @@
             icon="download"
             @click="download"
           />
+          <IconButton
+            v-if="step?.end_time !== undefined && user?.admin"
+            :title="$t('repo.pipeline.actions.log_delete')"
```

Comment:
As the api endpoint is required to have `session.MustPush,` this should check if the user has push access to the repo instead of `user.admin` which checks if the user is an instance admin.

Reply:
So, what direction it should be changed in? Check if the user has push access to the repo on frontend or require admin permissions on backend?

Reply:
IMO it would be best to allow it to repo admins.